### ISSUE TITLE
HTTP Span Attributes: `http.url` must not contain username / password

### ISSUE THREAD

**Issue by pellared** [open-telemetry/opentelemetry-java#3002]
As is stated in the recent [specification change](https://github.com/open-telemetry/opentelemetry-specification/pull/1502) :

> `http.url` MUST NOT contain credentials passed via URL in form of `https://username:password@www.example.com/`. In such case the attribute's value should be `https://www.example.com/`

**Comment by mateuszrzeszutek** [comment]
I don't think that otel-java contains any usage of the `http.url` attribute -- the [javaagent](https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/2535) is the correct place for this issue.

**Comment by pellared** [comment]
@mateuszrzeszutek I think [this example](https://github.com/open-telemetry/opentelemetry-java/blob/04fbc2e6c52100eff0ada86d6df61adbc8d09569/examples/http/src/main/java/io/opentelemetry/example/http/HttpClient.java#L62 ) could be reviewed just to make sure that others do not copy-paste potentially unsafe code.

**Comment by Hangzhi** [comment]
@jkwatson  @pellared Hello, I'm a student applying for Outreachy 2021 and in the process of contributing to the community. This issue seems like a good start. Could you please assign this issue to me?
